src/index.ts

```diff
@@ -142,54 +142,64 @@ export async function* getDeployedEntitiesStream(
+    //    in the range we are interested (>= genesisTimestamp)
+    if (
+      hash && lastIncludedDeploymentTimestamp && lastIncludedDeploymentTimestamp > genesisTimestamp &&
+      !(await components.processedSnapshotStorage.wasSnapshotProcessed(hash, replacedSnapshotHashes))) {
+      try {
+        // 2.1. download the snapshot file if needed
+        await downloadFileWithRetries(
+          components,
+          hash,
+          options.tmpDownloadFolder,
+          [options.contentServer],
+          new Map(),
+          options.requestMaxRetries,
+          options.requestRetryWaitTime
+        )
 
-      // 2.2. open the snapshot file and process line by line
-      const deploymentsInFile = processDeploymentsInFile(hash, components)
-      for await (const rawDeployment of deploymentsInFile) {
-        const deployment = coerceEntityDeployment(rawDeployment)
-        if (!deployment) continue
-        // selectively ignore deployments by localTimestamp
-        if (deployment.localTimestamp >= genesisTimestamp) {
-          components.metrics.increment('dcl_entities_deployments_processed_total', metricLabels)
-          yield deployment
-        }
-        // update greatest processed timestamp
-        if (deployment.localTimestamp > greatestProcessedTimestamp) {
-          greatestProcessedTimestamp = deployment.localTimestamp
+        // 2.2. open the snapshot file and process line by line
+        const deploymentsInFile = processDeploymentsInFile(hash, components)
+        for await (const rawDeployment of deploymentsInFile) {
+          const deployment = coerceEntityDeployment(rawDeployment)
+          if (!deployment) continue
+          // selectively ignore deployments by localTimestamp
+          if (deployment.localTimestamp >= genesisTimestamp) {
+            components.metrics.increment('dcl_entities_deployments_processed_total', metricLabels)
+            yield deployment
+          }
+          // update greatest processed timestamp
+          if (deployment.localTimestamp > greatestProcessedTimestamp) {
+            greatestProcessedTimestamp = deployment.localTimestamp
+          }
         }
-      }
-    } finally {
-      if (options.deleteSnapshotAfterUsage !== false) {
-        try {
-          await components.storage.delete([hash])
-        } catch (err: any) {
-          logs.error(err)
+        await components.processedSnapshotStorage.markSnapshotProcessed(hash, replacedSnapshotHashes)
+      } finally {
+        if (options.deleteSnapshotAfterUsage !== false) {
+          try {
+            await components.storage.delete([hash])
+          } catch (err: any) {
+            logs.error(err)
+          }
         }
       }
     }
+    greatestProcessedTimestamp = lastIncludedDeploymentTimestamp > greatestProcessedTimestamp
+      ? lastIncludedDeploymentTimestamp : greatestProcessedTimestamp
   }
 
+  logs.info(`End processing snapshots. Greatest processed timestamp: ${greatestProcessedTimestamp}`)
```

Comment:
Prefer logs without string interpolations
```suggestion
  logs.info(`End processing snapshots.`, {greatestProcessedTimestamp})
```